src/crypto/x509/x509.go

```diff
@@ -855,7 +868,7 @@ func checkSignature(algo SignatureAlgorithm, signed, signature []byte, publicKey
 
 	switch hashType {
 	case crypto.Hash(0):
-		if pubKeyAlgo != Ed25519 {
+		if pubKeyAlgo != Ed25519 && pubKeyAlgo != EdDilithium3 {
```

Comment:
Perhaps: `pubKeyAlgo != Ed25519 || pubKeyAlgo != EdDilithium3`

Reply:
(You've commented on an outdated commit.  In the latest version EdDilithium3 does not appear directly but rather the scheme is used.)

Direct signing should only be used with Ed25519 or EdDilithium3.  So it's ok when `pubKeyAlgo == Ed25519 || pubKeyAlgo == EdDiltihium3`.  Negating and pulling through the negation turns the or into an and.

Reply:
Gotcha.